### DISCUSSION TITLE
Comparison with OctoPrint-TPLinkSmartplug?

### DISCUSSION THREAD

**Discussion by debren27** [jamesmccannon02/OctoPrint-Tplinkautoshutdown#4] [Ideas]
Hi, how is this different from / better than https://github.com/jneilliii/OctoPrint-TPLinkSmartplug ?

I'm already using that to control power to my printer and supporting devices, including automatically turning them off.

Thanks!
Donovan

**Comment by CodingTheUnknown** [comment]
When connecting to the SmartPlug, it uses the Kasa library rather than the reversed engineered method used in the other plugin. This will allow this to become a more feature rich plugin as I develop it but also easier to maintain and keep bug free moving into the future as the kasa library is always being updated to support the newer smart plugs. 

I have been in contact with the maintainer of the other plugin and he has commented that this is a more efficient plugin to use. To help me though, it would be great if you use mine and then give feedback and features you want to see as you have experience with both

**Reply by CodingTheUnknown** [reply]
It’s also worth mentioning that I don’t use code to start the shutdown process but instead use event the listener which removes a further element of complexity

**Reply by debren27** [reply]
Very cool, thanks for responding! I'll have to give it a try.

One feature that gives me a warm fuzzy is auto-off on printer error, so that power is cut in case of thermal runaway. Also power strip support as I have everything on an HS300 and a KP303. (I control the printer, separate bed heater, LED lights, Mosaic Palette, OctoPi, dehydrator, and ventilation fan; half on UPS and half not.) I hope to see these features in your plugin!

Thank you for your work to make my 3D printing better!

**Reply by CodingTheUnknown** [reply]
Definitely will look at implementing these features. Especially the on error.
